connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -173,32 +186,54 @@ object UCSingleCatalog {
   val LOAD_DELTA_CATALOG = ThreadLocal.withInitial[Boolean](() => true)
   val DELTA_CATALOG_LOADED = ThreadLocal.withInitial[Boolean](() => false)
 
-  def generateCredentialProps(
-      scheme: String,
-      temporaryCredentials: TemporaryCredentials): Map[String, String] = {
+  private def createPathBasedCredProps(scheme: String, unityCatalogURL: String, unityCatalogToken: String, path: String, pathOp: PathOperation, tempCred: TemporaryCredentials): Map[String, String] = {
+    val props = commonProps(scheme, unityCatalogURL, unityCatalogToken, tempCred)
+
+    // Add keys for the path based temporary credential requests.
+    props ++ Map(
+      Constants.UNITY_CATALOG_PATH -> path,
+      Constants.UNITY_CATALOG_PATH_OPERATION -> pathOp.getValue,
+      Constants.UNITY_CATALOG_CREDENTIALS_TYPE -> Constants.UNITY_CATALOG_PATH_CREDENTIALS_TYPE
+    )
+  }
+
+  def createTableBasedCredProps(scheme: String, unityCatalogURL: String, unityCatalogToken: String, table: String, tableOp: TableOperation, tempCred: TemporaryCredentials): Map[String, String] = {
+    val props = commonProps(scheme, unityCatalogURL, unityCatalogToken, tempCred)
+
+    // Add keys for the path based temporary credential requests.
+    props ++ Map(
+      Constants.UNITY_CATALOG_TABLE -> table,
+      Constants.UNITY_CATALOG_TABLE_OPERATION -> tableOp.getValue,
+      Constants.UNITY_CATALOG_CREDENTIALS_TYPE -> Constants.UNITY_CATALOG_TABLE_CREDENTIALS_TYPE
+    )
+  }
+
+  private def commonProps(scheme: String, unityCatalogURL: String, unityCatalogToken: String, tempCred: TemporaryCredentials): Map[String, String] = {
     if (scheme == "s3") {
-      val awsCredentials = temporaryCredentials.getAwsTempCredentials
+      val awsCredentials = tempCred.getAwsTempCredentials
       Map(
-        // TODO: how to support s3:// properly?
         "fs.s3a.access.key" -> awsCredentials.getAccessKeyId,
         "fs.s3a.secret.key" -> awsCredentials.getSecretAccessKey,
         "fs.s3a.session.token" -> awsCredentials.getSessionToken,
+        Constants.UNITY_CATALOG_URI -> unityCatalogURL,
+        Constants.UNITY_CATALOG_TOKEN -> unityCatalogToken,
+        "fs.s3a.aws.credentials.provider" -> classOf[AwsVendedTokenProvider].getName,
```

Comment:
there is risk that in some environment this new stuff can cause errors while loading the provider. lets use a separate spark conf to enable the renew for now. we can always set it to be enable by default by changing the default value of the flag after we have gained confidence.

Reply:
> there is risk that in some environment this new stuff can cause errors while loading the provider

Agreed, there is risk.   If someone encounter this error and if he/she want to acess the UC,  then the only way that I can think of is to fix it, because finally they need this `AwsVendedTokenProvider` to create the temporary credentials.  There is no other way.  If this is correct, then seem no neccessary to introduce another switch to control enable the `AwsVendedTokenProvider` or not.  

Please correct me if I misunderstood, thanks for the comment !